packages/code-infra/src/utils/build.mjs

```diff
@@ -63,6 +66,29 @@ export function validatePkgJson(packageJson, options = {}) {
     }
   }
 
+  const reactVersion = packageJson.peerDependencies?.react;
+  if (enableReactCompiler) {
+    if (!reactVersion) {
+      errors.push(
+        'When building with React compiler, "react" must be specified as a peerDependency in package.json.',
+      );
+    }
+    const minSupportedReactVersion = semver.minVersion(reactVersion);
+    if (!minSupportedReactVersion) {
+      errors.push(
+        `Unable to determine the minimum supported React version from the peerDependency range: "${reactVersion}".`,
+      );
+    } else if (
+      semver.lt(minSupportedReactVersion, '19.0.0') &&
+      !packageJson.peerDependencies?.['react-compiler-runtime'] &&
```

Comment:
alternatively, the builder could add the dependency if it isn't specified? Not sure if it's a good idea